src/core/api/public_api.ts

```diff
@@ -711,50 +721,100 @@ class Player extends EventEmitter<IPublicAPIEvent> {
+     * a HTTP request to respond.
+     */
+    const resetState = () => {
+      this.stop();
+      this._priv_currentError = null;
+      this._priv_playing$.next(false);
+    };
 
     const videoElement = this.videoElement;
 
     /** Global "clock" used for content playback */
     const clock$ = createClock(videoElement, { withMediaSource: !isDirectFile,
                                                lowLatencyMode });
 
-    /** Emit when the current content has been stopped. */
-    const contentIsStopped$ = observableMerge(
-      this._priv_stopCurrentContent$,
-      this._priv_stopAtEnd ? onEnded$(videoElement) :
-                             EMPTY
-    ).pipe(take(1));
-
     /** Emit playback events. */
     let playback$ : ConnectableObservable<IInitEvent>;
 
     if (!isDirectFile) {
       const transportFn = features.transports[transport];
       if (typeof transportFn !== "function") {
+        resetState();
         throw new Error(`transport "${transport}" not supported`);
       }
 
       const transportPipelines = transportFn(transportOptions);
 
+      const { offlineRetry, segmentRetry, manifestRetry } = networkConfig;
+
+      /** Interface used to load and refresh the Manifest. */
+      const manifestFetcher = new ManifestFetcher(url,
+                                                  transportPipelines,
+                                                  { lowLatencyMode,
+                                                    maxRetryRegular: manifestRetry,
+                                                    maxRetryOffline: offlineRetry });
+
+      /** Interface used to download segments. */
+      const segmentFetcherCreator = new SegmentFetcherCreator<any>(
+        transportPipelines,
+        { lowLatencyMode,
+          maxRetryOffline: offlineRetry,
+          maxRetryRegular: segmentRetry });
+
+      /** Observable emitting the initial Manifest */
+      let manifest$ : Observable<IManifestFetcherParsedResult |
+                                 IManifestFetcherWarningEvent>;
+
+      if (initialManifest instanceof Manifest) {
+        manifest$ = observableOf({ type: "parsed",
+                                   manifest: initialManifest });
+      } else if (initialManifest !== undefined) {
+        manifest$ = manifestFetcher.parse(initialManifest, { previousManifest: null,
+                                                             unsafeMode: false });
+      } else {
+        manifest$ = manifestFetcher.fetch(url).pipe(
+          mergeMap((response) => response.type === "warning" ?
+            observableOf(response) : // bubble-up warnings
+            response.parse({ previousManifest: null, unsafeMode: false })));
+      }
+
+      // Load the Manifest right now and share it with every subscriber until
+      // the content is stopped
+      manifest$ = manifest$.pipe(takeUntil(stopContent$),
+                                 shareReplay());
+      manifest$.subscribe();
```

Comment:
Do you think it will be less efficient to subscribe `manifest$` only when `playback$` is subscribed, later in the code?

Reply:
Well that's kind of the point of that PR, to not do that :p

Basically we want to do it before the `stop()` call, which might take some (synchronous) time and before the contentLock$ is freed.